res/qml/Mixxx/Controls/WaveformOverviewMarker.qml

```diff
@@ -42,7 +43,10 @@ Item {
 
         group: root.group
         key: root.key
-        onValueChanged: marker.x = parent.width * value
+        onValueChanged: {
+            // Math.round saves tons of CPU by avoiding redrawing for fractional pixel positions.
+            marker.x = Math.round(parent.width * value * Screen.devicePixelRatio) / Screen.devicePixelRatio;
```

Comment:
Yes, but only for fractional logical pixels that correspond to real display pixels. So for a scale factor of 2, updates are done for every 0.5 logical pixels, or every real display pixel, but not extremely tiny fractions in between.

Reply:
Ah, so I guess the speedup comes from the fact that Qt doesn't have to do anti aliasing?

Reply:
Perhaps?

Reply:
That doesn't make sense either though.

This feels like it should not be something programming QML should have to concern themselves with. So I feel like this is bug in Qt?

Reply:
I don't know, but I know that this little change saves a lot of CPU :shrug:

Reply:
Mhmm the only other idea I have is that the COs value changed multiple times per frame so the callback is also executed quite often. Since the position is always slightly different, the return value is also different. When rounding we always return the same value in the same pixel, so we often return the same value which enables Qt to do some caching optimization (so only doing work when the value actually changes).